diff --git a/main.c b/main.c
index 593e1c1..4722cda 100644
--- a/main.c
+++ b/main.c
@@ -573,6 +573,7 @@ void on_keypress(XKeyEvent *kev)
 	unsigned int sh = 0;
 	KeySym ksym, shksym;
 	char dummy, key;
+	bool bound = false;
 	bool dirty = false;
 
 	XLookupString(kev, &key, 1, &ksym, NULL);
@@ -590,6 +591,7 @@ void on_keypress(XKeyEvent *kev)
 		extprefix = False;
 	} else if (extprefix) {
 		run_key_handler(XKeysymToString(ksym), kev->state & ~sh);
+                bound = true;
 		extprefix = False;
 	} else if (key >= '0' && key <= '9') {
 		/* number prefix for commands */
@@ -603,8 +605,11 @@ void on_keypress(XKeyEvent *kev)
 		{
 			if (cmds[keys[i].cmd].func(keys[i].arg))
 				dirty = true;
+			bound = true;
 		}
 	}
+	if (!bound)
+		run_key_handler(XKeysymToString(ksym), kev->state & ~sh);
 	if (dirty)
 		redraw();
 	prefix = 0;
